source/extensions/filters/http/common/aws/credentials_provider.h

```diff
@@ -13,17 +13,27 @@ namespace HttpFilters {
 namespace Common {
 namespace Aws {
 
+/**
+ * AWS credentials container
+ *
+ * If a credential component was not found in the execution environment, it's getter method will
+ * return absl::nullopt. Credential components with the empty string value are treated as not found.
+ */
 class Credentials {
 public:
-  Credentials() = default;
-
-  Credentials(absl::string_view access_key_id, absl::string_view secret_access_key)
-      : access_key_id_(access_key_id), secret_access_key_(secret_access_key) {}
-
-  Credentials(absl::string_view access_key_id, absl::string_view secret_access_key,
-              absl::string_view session_token)
-      : access_key_id_(access_key_id), secret_access_key_(secret_access_key),
-        session_token_(session_token) {}
+  Credentials(absl::string_view access_key_id = absl::string_view(),
+              absl::string_view secret_access_key = absl::string_view(),
+              absl::string_view session_token = absl::string_view()) {
+    if (!access_key_id.empty()) {
```

Comment:
Does this mean we always need `access_key_id`? and we require either `secret_access_key` or `session_token` to be there? This if statements seem not necessary. What is the implication if we don't have (or empty) `access_key_id`?

Reply:
Credentials can be one of:

- Empty
- Have access key only
- Have access key and secret key
- Have access key, secret key and session token

Implication of an empty access key is that request will be made with anonymous identity. For example requests can be made to S3 with anonymous identity if bucket is configured in that way. Non-production environments can also accept anonymous requests.